typescript/hx-adl/src/gen-sqlschema.ts

```diff
@@ -72,128 +86,185 @@ export async function generateSql(params: Params): Promise<void> {
   const loadedAdl = await parseAdl(params.adlFiles, params.adlSearchPath);
 
   // Find all of the struct declarations that have a DbTable annotation
-  const dbTables: DbTable[]  = [];
+  const dbTables: DbTable[] = [];
 
   forEachDecl(loadedAdl.modules, scopedDecl => {
     if (scopedDecl.decl.type_.kind == 'struct_') {
       const struct = scopedDecl.decl.type_;
       const ann = getAnnotation(scopedDecl.decl.annotations, DB_TABLE);
       if (ann != undefined) {
         const name = getTableName(scopedDecl);
-        dbTables.push({scopedDecl, struct, ann, name});
+        dbTables.push({ scopedDecl, struct, ann, name });
       }
     }
   });
-  dbTables.sort( (t1, t2) => t1.name < t2.name ? -1 : t1.name > t2.name ? 1 : 0);
+  dbTables.sort((t1, t2) => t1.name < t2.name ? -1 : t1.name > t2.name ? 1 : 0);
   await generateSqlSchema(params, loadedAdl, dbTables);
   if (params.outmetadata !== null) {
     await generateMetadata(params.outmetadata, params, loadedAdl, dbTables);
   }
-  for(const t of params.templates) {
+  for (const t of params.templates) {
     await generateTemplate(t, dbTables);
   }
 }
 
 async function generateSqlSchema(params: Params, loadedAdl: LoadedAdl, dbTables: DbTable[]): Promise<void> {
   // Now generate the SQL file
-  const writer = fs.createWriteStream(params.outfile);
-  const moduleNames : Set<string> = new Set(dbTables.map(dbt => dbt.scopedDecl.moduleName));
-  writer.write( `-- Schema auto-generated from adl modules: ${Array.from(moduleNames.keys()).join(', ')}\n` );
-  writer.write( `--\n` );
-  writer.write( `-- column comments show original ADL types\n` );
+  let writer = fs.createWriteStream(params.outfile);
+  const moduleNames: Set<string> = new Set(dbTables.map(dbt => dbt.scopedDecl.moduleName));
+  writer.write(`-- Schema auto-generated from adl modules: ${Array.from(moduleNames.keys()).join(', ')}\n`);
+  writer.write(`--\n`);
+  writer.write(`-- column comments show original ADL types\n`);
 
   if (params.extensions.length > 0) {
     writer.write('\n');
-    params.extensions.forEach( e => {
-      writer.write( `create extension ${e};\n` );
+    params.extensions.forEach(e => {
+      writer.write(`create extension ${e};\n`);
     });
   }
 
+  const collectSetNotNullLines: { table: string, cols: string[] }[] = [];
   const constraints: string[] = [];
   let allExtraSql: string[] = [];
 
+  writer.write("begin;\n");
```

Comment:
It is not necessarily correct to write `begin` and `commit`

There are use cases for having them and for not having them.
It could be an option to add in perhaps.